diarymanagement-code-r273/tbaclient/trunk/src/TBA/Sounds/SoundResources.java

```diff
@@ -0,0 +1,119 @@
+/*
+ * To change this template, choose Tools | Templates
+ * and open the template in the editor.
+ */
+
+package TBA.Sounds;
+
+import java.io.*;
+import javax.sound.sampled.*;
+
+/**
+ *  The SoundPlayer class loads wav files into memory and plays them back when needed
+ *
+ * @author cs321tx2
+ */
+public class SoundResources
+{
+
+    boolean isNowMute;
+
+    /** now field is file name of sound that plays back when cell is red */
+    String nowLocation = "defaultNow.wav"; //atm this is just windows' chime.wav renamed
+    
+    //String soonLocation = "defaultSoon.wav";
+    String soonLocation = "defaultNow.wav";
+    /**
+     * actual sound loaded in memory; to play just SoundPlayer.nowClip.start();
+     */
+    public Clip nowClip;
+
+    /**
+     * constructor loads sounds into memory
+     */
+    public SoundResources()
+    {
+        isNowMute = false;
+        nowClip = Load(nowLocation);
+    }
+
+    public void Reload()
+    {
+        nowClip = Load(nowLocation);
+    }
+
+    /**
+     *private function called by constructor that loads sound file into memory
+     */
+    private Clip Load(String filename)
+    {
+        try
+        {
+            //open file
+            //File file = new File(filename);
+            AudioInputStream stream = AudioSystem.getAudioInputStream(SoundResources.class.getResourceAsStream(filename));
+            //AudioInputStream stream = AudioSystem.getAudioInputStream(file);
+            AudioFormat format = stream.getFormat();
+
+            //we're using clips so that we can just play back sounds from mmemory
+            DataLine.Info lineInfo = new DataLine.Info(Clip.class, format);
+
+            //make line
+            Clip aClip = (Clip)AudioSystem.getLine(lineInfo);
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer

**Problem**
This line of code contains a resource that might not be closed properly. Resource leaks can cause your system to slow down or crash.

**Fix**
Consider closing the resource returned by the following method call: *getLine*. The resource is referenced in statements at the following lines: 62, 65. The resource is returned at line: 67. There are other execution paths that neither close the resource nor return it, e.g., when exception is thrown by Clip.open. Either a) close the object returned by *getLine()* in a try-finally block or b) close the resource by declaring the object returned by *getLine()* in a try-with-resources block.

**More info**
[View resource management guidelines at oracle.com](https://www.oracle.com/technetwork/articles/java/trywithresources-401775.html) (external link).